### PR TITLE
[TRANSFORMATIONS -> CONSTANT FOLDING] Add evaluate for SqDiff due to using by constant folding

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *This pattern was not applied by `Constant folding transformation:`*
image
- The result after constant folding before fix:
image
- The result after constant folding after fix:
image

### Tickets:
 - *https://jira.devtools.intel.com/browse/CVS-129152*

### CODE REVIEW

**Review by itikhono — APPROVED** [review]
Looks good!
Waiting for a review from Core team @praasz

**Review by mateusztabaka — APPROVED** [review]
you can consider adding some tests to src/core/tests/pass/constant_folding.cpp

File: src/core/src/op/squared_difference.cpp
Comment (itikhono) [inline src/core/src/op/squared_difference.cpp:70]: minor:
can we just check the lib bin size before/after adding f32, f16,i16,i32 etc types here?
It's just interesting point

**Review by praasz — APPROVED** [review]

File: src/core/src/op/squared_difference.cpp
Comment (itikhono) [reply src/core/src/op/squared_difference.cpp:74]: let's do this in the next PR.

**Pull request merged**
